src/tools/clippy/tests/ui/or_fun_call.stderr

```diff
@@ -240,12 +240,6 @@ error: use of `unwrap_or` to construct default value
 LL |     let _ = opt.unwrap_or({ i32::default() });
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `unwrap_or_default()`
 
-error: function call inside of `unwrap_or`
-  --> tests/ui/or_fun_call.rs:409:21
-   |
-LL |     let _ = opt_foo.unwrap_or(Foo { val: String::default() });
-   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `unwrap_or_else(|| Foo { val: String::default() })`
```

Comment:
Should this clippy lint have a check that the value is const constructable in a stable way (which would revert this output change)?

Reply:
Isn't it the other way around: users don't get a lint to do `unwrap_or_else(NowConstThing)` because `NowConstThing` is now detected as const by clippy, even if it isn't const in stable? So it's not that the users get unactionable feedback, but rather that they don't get feedback actionable under older versions of the std. Does rustc take advantage of things that are unstably const for const folding on stable?